ocaml/typing/uniqueness_analysis.ml

```diff
@@ -995,10 +1012,13 @@ let conjuncts_pattern_match l =
 let rec pattern_match_tuple pat values =
   match pat.pat_desc with
   | Tpat_or (pat0, pat1, _) ->
+    Unique_barrier.enable pat.pat_unique_barrier;
```

Comment:
Could/should this step happen in `pattern_match`?

Reply:
No, because `pattern_match` only sees the topmost or-pattern, but `pat0` and `pat1` might be or-patterns themselves